R/library.R

```diff
@@ -272,6 +279,20 @@ download_package.github <- function(package, force) {
   remote_download.github_remote(remote, dest, quiet = !isTRUE(getOption('lockbox.verbose')))
 }
 
+## download a raw tarball from a URL
+download_package.tarball <- function(package, force) {
+  dest <- lockbox_package_download_path(package)
+  if (is.na(package$version)) {
+    package$version <- NULL
+  }
+
+  suppressWarnings(tryCatch(
+    download.file(url = package$url, destfile = lockbox_package_download_path(package)
+    , quiet = notTRUE(getOption('lockbox.verbose'))), error = function(e) e))
+
```

Comment:
Maybe double check here that the downloaded file is in fact a tarball.

Reply:
how would I do that? I'm effectively downloading a binary from a URL I specified in the yaml file and hope that it's a tarball